### PR TITLE
DOC Document how to enable the preview panel for custom DataObjects.

### PR DESCRIPTION
**Pull request by GuySartorelli**
Documentation for enabling the preview panel for custom `DataObject`s.

Depends on the other pull requests that are listed in silverstripe/silverstripe-admin#1251. If any of those pull requests change in a material way, this documentation will need to be updated.

### CODE REVIEW

**Comment by emteknetnz** [comment]
@GuySartorelli is this ready for review?

**Comment by GuySartorelli** [comment]
> @GuySartorelli is this ready for review?

Not yet. It's got some code examples of how to make a `DataObject` previewable under various circumstances, but I want to add a few words around the code to provide context.
I also want to add a small change to the docs at https://userhelp.silverstripe.org/en/4/creating_pages_and_content/creating_and_editing_content/previewing_changes/ to point out the new unversioned preview state.

I'm hoping to get to it this week, but will be done mid next week at the latest. Let me know if that is too slow and I'll see what I can do to move my priorities around.

**Comment by GuySartorelli** [comment]
@emteknetnz I'm starting to work on this now - do you know where the source for https://userhelp.silverstripe.org/en/4/creating_pages_and_content/creating_and_editing_content/previewing_changes/ is? It doesn't appear to be in the docs folder in this repo.

**Comment by emteknetnz** [comment]
https://github.com/silverstripe/silverstripe-userhelp-content

**Comment by GuySartorelli** [comment]
@emteknetnz This is ready for review now. I'll open a PR for the userhelp docs now so that I can link to it from the overarching issue.

I'm not sure that deprecation notice in the javascript is helpful, and if you agree, I'd like to remove it. There's no react replacement for the preview javascript API to my knowledge, so this notice effectively tells developers "You should use the react preview component that doesn't exist instead."

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md
Comment (GuySartorelli) [inline docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md:93]:
```suggestion
state you can use. This state will always be active once you enable it.
```

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md
Comment (GuySartorelli) [inline docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md:108]:
```suggestion
It also needs at least one preview state enabled. This example assumes we aren't using
```

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md
Comment (GuySartorelli) [inline docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md:176]:
```suggestion
list of items and indirectly calling `forTemplate` using the [`$Me` template variable](../01_Templates/01_Syntax.md#me).
This method will be used by the `MyAdmin` class in the `cmsPreview` action to tell the CMS what to display in the preview panel.
```

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md
Comment (GuySartorelli) [inline docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md:249]:
```suggestion
edit form would be loaded instead of the `DataObject`'s form. To avoid this unexpected behaviour, you
```

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md
Comment (GuySartorelli) [inline docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md:250]:
```suggestion
can include an extra GET parameter in the `PreviewLink`. Then in the `MetaTags` method,
```

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md
Comment (GuySartorelli) [inline docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md:340]:
```suggestion
 * Used to generated the id for the product element in the template.
```
Oops, this was a copy/paste error. This concept was taken from the elemental module so there may be a couple other holdovers from that which I missed.

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md
Comment (GuySartorelli) [inline docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md:376]:
```suggestion
<% loop $Products ‰>
```

**Comment by GuySartorelli** [comment]
@emteknetnz I decided to read back through it and found a few minor issues I missed while writing it. I'll hold off actually making these changes until you or another core team member has given this a review.

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md
Comment (emteknetnz) [inline docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md:312]:
```suggestion
        'ProductPage' => ProductPage::class,
```

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md
Comment (emteknetnz) [inline docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md:356]:
```suggestion
    if ($page = $this->ProductPage()) {
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Fantastic work.

I've tested the code samples on my local, a few small changes need to be made then we're good to merge

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md
Comment (GuySartorelli) [reply docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md:67]: Yup I thought that might need to be removed 😅
Might be worth looking at deprecating that method and removing it in 5, but that's out of scope for now.

**Comment by GuySartorelli** [comment]
@emteknetnz Pretty sure I got all those changes. :)

File: docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md
Comment (GuySartorelli) [reply docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md:208]: Oops, missed the menu title. Will do that now

**Comment by emteknetnz** [comment]
Still need `private static $menu_title = 'Products';` :-)

**Review by emteknetnz — APPROVED** [review]
Thanks!

**Comment by GuySartorelli** [comment]
Thank you! It's nice to have this whole thing merged in. 🎉

**Pull request merged**
